spec/components/provider_interface/conditions_form_component_spec.rb

```diff
@@ -14,7 +14,7 @@
   let(:further_conditions) { [] }
   let(:further_condition_models) do
     further_conditions.map.with_index do |condition, index|
-      OpenStruct.new(id: index, text: condition)
+      Struct.new(:id, :condition_id, :text).new(index, index, condition)
```

Comment:
I was wondering why we have both id and condition id here?

Reply:
`condition_id` is required by the form and if not set it results in an error. 

It should actually be set to nil so I'll update this. This is because of the slightly different behavior between a Struct and an OpenStruct, as OpenStruct will return nil when attempting to access a non defined attribute, but Struct will raise a `NoMethodError`.